pluma/src/main/java/com/flipstudio/pluma/Database.java

```diff
@@ -195,6 +201,74 @@ public void registerFunction(String name, int numArgs, SQLiteFunction function)
 			throw new SQLiteException(rc, lastErrorMessage(mDB));
 		}
 	}
+
+	public void executeActionAfterCommit(Action0 action) {
+		if (!isInTransaction()) {
+			action.call();
+		} else {
+			if (mActions == null) mActions = new Array<>();
+			mActions.add(action);
+		}
+	}
+
+	public void beginTransaction() {
+		try {
+			executeUpdate("BEGIN");
+			mIsInTransaction++;
+		} catch (SQLiteException e) {
+			throw new RuntimeException(e.toString());
+		}
+	}
+
+	public void commitTransaction() {
+		try {
+			executeUpdate("COMMIT");
+			mIsInTransaction--;
+		} catch (SQLiteException e) {
+			throw new RuntimeException(e.toString());
+		}
+
+		if (!isInTransaction()) {
+			for (final Action0 action : mActions) {
+				action.call();
+			}
+		}
+	}
+
+	public void rollbackTransaction() {
+		try {
+			executeUpdate("ROLLBACK");
+			mIsInTransaction--;
+		} catch (SQLiteException e) {
+			throw new RuntimeException(e.toString());
+		}
+	}
+
+	public void executeWithinTransaction(Action0 action) {
+		try {
+			beginTransaction();
+			action.call();
+			commitTransaction();
+		} catch (Exception e) {
+			rollbackTransaction();
+			throw new RuntimeException(e.toString());
```

Comment:
http://stackoverflow.com/questions/1452790/will-database-file-of-sqlite3-be-damaged-when-suddenly-power-off-or-os-crash não vai completar a transaction